### PR TITLE
Add GSL Stanage

### PR DESCRIPTION
**Pull request by Carldkennedy**
Adds GSL documentation for Stanage

### CODE REVIEW

File: stanage/software/libs/gsl.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+   :Documentation: https://www.gnu.org/software/gsl/doc/html/index.html
+
+The GNU Scientific Library (GSL) is a collection of routines for numerical computing. 
+The routines have been written from scratch in C.  
+See `here <https://www.gnu.org/software/gsl/doc/html/intro.html>`__ for the types of routines that the GSL provides.
+
+Usage
+-----
+
+The GSL library can be loaded by running one of: ::
```
Comment (willfurnass) [inline stanage/software/libs/gsl.rst:22]: Note that only have icelake builds at present?

**Review by willfurnass — APPROVED** [review]
LGTM.  See minor comment.

File: stanage/software/libs/gsl.rst [hunk 2]
```diff
@@ -19,7 +19,13 @@ See `here <https://www.gnu.org/software/gsl/doc/html/intro.html>`__ for the type
 Usage
 -----
 
-The GSL library can be loaded by running one of: ::
+.. note::
+
```
Comment (willfurnass) [inline stanage/software/libs/gsl.rst:23]: Is this consistent with how we've handled this elsewhere?  should users know what 'Icelake' is before arriving at this page?

Comment (willfurnass) [reply stanage/software/libs/gsl.rst:23]: Perhaps we don't need to resolve this now, but I think it's worth thinking about what Research Support want the general approach to be on Stanage; I think at present there's the risk of confusion on GPU nodes regarding what's installed.

Comment (Carldkennedy) [reply stanage/software/libs/gsl.rst:23]: We could be indicate CPU and/or GPU node in the sidebar?

Using tabs, where available modules are different on each architecture?

Comment (willfurnass) [reply stanage/software/libs/gsl.rst:23]: @nikishe and @desmfryan Thoughts?

Comment (Carldkennedy) [reply stanage/software/libs/gsl.rst:23]: See #2087 for an example

**Review by nikishe — APPROVED** [review]

**Review by nikishe — APPROVED** [review]
Good work Carl

File: stanage/software/libs/gsl.rst (hunk 2 above)
Comment (nikishe) [reply stanage/software/libs/gsl.rst:20]: Tabs to separate GPU/cpu seems good. You can have it as inner tabs as well as long as its not to squashed

**Pull request merged**